### ISSUE TITLE
Question about Lint/ToEnumArguments

### ISSUE THREAD

**Issue by bquorning** [rubocop/rubocop#12885]
Maybe it’s a bug, or maybe I just don’t understand the cop.

The [cop’s spec](https://github.com/rubocop/rubocop/blame/7f9595aa3ca3524f888b1464c4eeb7e15aabfc22/spec/rubocop/cop/lint/to_enum_arguments_spec.rb#L107-L114) includes this example:

```rb
  it 'registers an offense when enumerator is created for another method' do
    expect_offense(<<~RUBY)
      def m(x)
        return to_enum(:not_m) unless block_given?
               ^^^^^^^^^^^^^^^ Ensure you correctly provided all the arguments.
      end
    RUBY
  end
```

How do I change the code to make the cop accept it? I tried `return to_enum(:not_m, x) unless block_given?` but still get the same offense.

The [cop documentation](https://docs.rubocop.org/rubocop/1.63/cops_lint.html#linttoenumarguments) says the cop

> Ensures that `to_enum`/`enum_for`, called for the current method, has correct arguments.

Since `not_m` is not “the current method” (which is `#m`), either the comment is wrong, or the cop should completely ignore the code in question.

## RuboCop version

```
$ bundle exec rubocop -V
1.63.4 (using Parser 3.3.1.0, rubocop-ast 1.31.2, running on ruby 3.3.1) [arm64-darwin22]
  - rubocop-performance 1.21.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.29.2
```

**Comment by vlad-pisanov** [comment]
It's a bug; this case should be ignored. We can't make assumptions about parameters that `:not_m` needs; it may not even depend on `x`.

```ruby
def m(x)
  return to_enum(:not_m, some_option: true) # should be allowed
  ...
end
```
